modules/stochastic_tools/src/samplers/MonteCarloSampler.C

```diff
@@ -19,43 +12,22 @@ InputParameters
 validParams<MonteCarloSampler>()
 {
   InputParameters params = validParams<Sampler>();
-  params.addRequiredParam<unsigned int>("n_samples", "Number of samples for Monte Carlo Sampling");
   params.addClassDescription("Monte Carlo Sampler.");
+  params.addRequiredParam<unsigned int>(
+      "n_samples", "Number of Monte Carlo samples to perform for each distribution.");
   return params;
 }
 
 MonteCarloSampler::MonteCarloSampler(const InputParameters & parameters)
   : Sampler(parameters), _num_samples(getParam<unsigned int>("n_samples"))
 {
-  /// initial samples of perturbed parameters
-  generateSamples();
 }
 
-MonteCarloSampler::~MonteCarloSampler() {}
-
-void
-MonteCarloSampler::generateSamples()
+DenseMatrix<Real>
+MonteCarloSampler::sampleDistribution(Distribution & distribution, unsigned int)
 {
-  _current_sample += 1;
-  Real weight = 1.0;
-  _var_value_map.clear();
-  for (auto & var_name : _var_names)
-  {
-    if (_reseed_for_new_sample)
-    {
-      unsigned int randomSeed = getRandomLong();
-      _var_dist_map[var_name]->setSeed(randomSeed);
-    }
-    Real val = _var_dist_map[var_name]->getRandomNumber();
-    _var_value_map[var_name] = val;
-    if (_var_value_hist.find(var_name) != _var_value_hist.end())
-      _var_value_hist[var_name].push_back(val);
-    else
-      mooseError("The variable: ",
-                 var_name,
-                 " is not found in the predetermined map, i.e. _var_value_hist");
-  }
-  /// The probability weight for Monte Carlo sampler is constant
-  /// This should be normalized before any data analysis of outputs
-  _probability_weight.push_back(weight);
+  DenseMatrix<Real> output(_num_samples, 1);
+  for (std::size_t i = 0; i < _num_samples; ++i)
```

Comment:
Well then you are missing the point. The chances of you getting all types to match when you are hardcoding them is basically zero (maybe even -2). I agree that _num_samples should probably be _size_t, but then a tight little iteration loop should just respect the type (i.e. `auto`). If it gets changed later, `auto` will always ensure that it's correct.